src/DurableTask.Generators/AzureFunctions/SyntaxNodeUtility.cs

```diff
@@ -57,55 +62,54 @@ public static bool TryGetFunctionKind(MethodDeclarationSyntax method, out Durabl
                 }
             }
 
-            kind = DurableFunctionKind.Unknonwn;
+            kind = DurableFunctionKind.Unknown;
             return false;
         }
 
-        public static bool TryGetRequiredNamespaces(SemanticModel model, List<TypeSyntax> types, out HashSet<string>? requiredNamespaces)
+        public static bool TryGetRequiredNamespaces(List<INamedTypeSymbol> types, out HashSet<string>? requiredNamespaces)
         {
             requiredNamespaces = new HashSet<string>();
 
-            var remaining = new Queue<TypeSyntax>(types);
+            var remaining = new Queue<INamedTypeSymbol>(types);
 
             while (remaining.Any())
             {
-                var toProcess = remaining.Dequeue();
-
-                if (toProcess is PredefinedTypeSyntax)
-                    continue;
-
-                TypeInfo typeInfo = model.GetTypeInfo(toProcess);
-                if (typeInfo.Type == null)
+                INamedTypeSymbol typeInfo = remaining.Dequeue();
+                if (typeInfo is null)
                 {
                     return false;
                 }
 
-                if (toProcess is not PredefinedTypeSyntax && typeInfo.Type.ContainingNamespace.IsGlobalNamespace)
+                if (typeInfo.ContainingNamespace.IsGlobalNamespace)
                 {
                     requiredNamespaces = null;
                     return false;
```

Comment:
I know you didn't write this code, but is it correct for us to bail like this if we encounter a global namespace? Why wouldn't we continue to process remaining types in the `remaining` list?